libtenzir/builtins/formats/cef.cpp

```diff
@@ -62,26 +62,45 @@ std::string unescape(std::string_view value) {
   return result;
 }
 
-/// Parses the CEF extension field as a sequence of key-value pairs for further
-/// downstream processing.
+/// Parses the CEF extension field as a sequence of key-value pairs.
+/// This works by finding the kv separator ('=') and then using everything
+/// between a (not quoted) whitespace and that separator as the key. The value
+/// is simply the text after the separator before the start of the next key.
 /// @param extension The string value of the extension field.
 /// @returns A vector of key-value pairs with properly unescaped values.
 auto parse_extension(std::string_view extension,
                      auto builder) -> std::optional<diagnostic> {
   if (extension.empty()) {
     return {};
   }
-  auto key = extension.substr(0, extension.find('='));
-  extension.remove_prefix(key.size() + 1);
+  // find the first not quoted, not escaped kv separator
+  auto key_end = detail::find_first_not_in_quotes(extension, '=');
+  while (key_end != extension.npos
+         and (key_end > 0 and extension[key_end - 1] == '\\')) {
+    key_end = extension.find('=', key_end + 1);
+  }
+  if (key_end == extension.npos) {
+    return diagnostic::warning(
+             "extension field did not contain a key-value separator")
+      .done();
+  }
+  // extract the first key
+  auto key = extension.substr(0, key_end);
+  extension.remove_prefix(key_end + 1);
+  if (extension.empty()) {
+    return diagnostic::warning("extension field did not contain a value").done();
+  }
+  // find the next not quoted, not escaped kv separator
   while (not extension.empty()) {
-    auto next_kv_sep = extension.find('=');
-    while (next_kv_sep != extension.npos
-           and extension[next_kv_sep - 1] == '\\') {
-      next_kv_sep = extension.find('=', next_kv_sep + 1);
+    key_end = detail::find_first_not_in_quotes(extension, '=');
+    while (key_end != extension.npos
+           and (key_end > 0 and extension[key_end - 1] == '\\')) {
+      key_end = extension.find('=', key_end + 1);
     }
-    auto value_end = next_kv_sep == extension.npos
+    // find the last whitespace, determining the end of the value text
+    auto value_end = key_end == extension.npos
                        ? extension.npos
-                       : extension.find_last_of(" \t", next_kv_sep);
+                       : extension.find_last_of(" \t", key_end);
```

Comment:
It would be part of the previous value. `data_unparsed`  would actually trim the trailing whitespace, but I suppose I can just explicitly trim it so the `parse_cef` function doesnt end up with trailing whitespace in a value.

Reply:
> `data_unparsed` would actually trim the trailing whitespace

Wait, really? Where, and why?

Reply:
Now that you are asking: I dont know. 

I originally had it trim whitespace and strip quotes, because a lot of formats (xsv, cef, leef) may have quoted values and potential leading/trailing whitespace, where the quotes are supposed to be stripped. However, its clearly a format specific thing and not something the builder should do in general. 

I've now moved this into the respective formats.